File: src/nodes/decrement.zig
const nd = @import("node.zig");

pub const NodeDecrement = struct {
    operand: *nd.Node,

    pub fn init(operand: *nd.Node) NodeDecrement {
        return NodeDecrement{ .operand = operand };
    }

    pub fn get_operand(self: *const NodeDecrement) *nd.Node {
        return self.operand;
    }
};

File: src/nodes/decrement.rs
use crate::nodes::node::Node;

pub struct NodeDecrement {
    operand: Node,
}

impl NodeDecrement {
    pub fn init(operand: Node) -> NodeDecrement {
        NodeDecrement { operand }
    }
    
    pub fn get_operand(&self) -> &Node {
        &self.operand
    }
}